packages/flutter/lib/src/material/scaffold.dart

```diff
@@ -610,9 +610,10 @@ class ScaffoldState extends State<Scaffold> with TickerProviderStateMixin {
     if (_snackBars.isEmpty || _snackBarController.status == AnimationStatus.dismissed)
       return;
     final Completer<SnackBarClosedReason> completer = _snackBars.first._completer;
-    if (!completer.isCompleted)
-      completer.complete(reason);
-    _snackBarController.reverse();
+    _snackBarController.reverse().then((_) {
+      if (!completer.isCompleted)
```

Comment:
Can you assert(mounted) here? (I'm assuming it's not possible for the animation controller to complete after the State has gone away.)